rfc-192-use-codeowners-in-all-repos.md

```diff
@@ -0,0 +1,77 @@
+review code changes and respond to issues.
+
+## Proposal
+
+For the purposes of this RFC, the term _active repository_ means a repository
+that is:
+
+1. within the `dxw` organisation on GitHub or the equivalent on GitLab,
+1. not a fork of a repository from outside the organisation; and
+1. not archived.
+
+All active repositories MUST have a `CODEOWNERS` file in the root of the
+repository. There SHOULD be more than two members of staff in each file, who
+SHOULD have different roles in the organisation, to avoid all owners of a
+repository being unavailable at once. To save on maintenance time it is
+RECOMMENDED that code owners be a GitHub team, rather than a list of
+individuals.
+
+`CODEOWNERS` files MAY specify different owners for different files, or file
+types, but this should be avoided if it is likely to make it more difficult to
+ensure that contributions are reviewed in a timely manner.
+
+The use of `CODEOWNERS` MUST NOT be used to discourage developers from
+commenting, reviewing or otherwise participating in code maintenance. This is
+particularly important for new or junior members of staff, who should be
+encouraged to participate in code maintenance and review, in a supportive
+environment that
+[assumes good faith](https://en.wikipedia.org/wiki/Wikipedia:Assume_good_faith).
+
+Repositories SHOULD be configured to ensure that code owners review PRs before
+they are merged into default or `main` branches.
+
+Code owners will share responsibility for ensuring that repositories are well
+maintained, and that PRs and issues receive a timely response. In particular,
+code owners SHOULD ensure that their active repositories:
+
+- have a `CODEOWNERS` file which conforms to this RFC,
+- have a licence, code of conduct, and contribution guide, which MAY just link
+  to the [dxw default community health files](https://github.com/dxw/.github),
+  as this repository does,
+- have a set of
+  [topics](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/classifying-your-repository-with-topics)
+  that indicate which team the repository belongs to, and help other developers
+  identify the intent of the repository,
+- follow all RFCs; and
+- are archived as soon as the repository is no longer used and maintained.
+
+## Next steps
+
+- [ ] Generate a data dump of all GitHub repositories within the `dxw`
+      organisation and identify those that can be archived immediately; related
+      to #149
+- [ ] Add `CODEOWNERS` files to all active repositories.
+- [ ] Ensure all repositories have
+      [topics](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/classifying-your-repository-with-topics)
+      set which identify the owner (e.g. `govpress`, `tech-ops`, etc.)
+- [ ] Write a tool to generate PRs to add a default license, `CONTRIBUTING.md`
```

Comment:
We've [done this before](https://github.com/dxw/repo-audit). There was definitely interest in turning it into an internal platform which would let us quickly get an overview, which would also align with @Floppy's point about a central "who owns this code" view.

Reply:
Yes, I've been looking through that repo. I'm beginning to think we need two things: 1) a relatively regular "audit" of the repos that highlights places where maintenance is needed, and 2) to start using that tool, or something very similar, to automatically resolve maintenance issues, like missing files.